File: tks/RepositoriesAdapters/src/test/kotlin/mappers/UserMapperTest.java
package mappers;

import data.UserEntity;
import lombok.SneakyThrows;
import model.User;
import model.UserRole;
import model.values.Email;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import java.util.UUID;

import static org.junit.jupiter.api.Assertions.*;

class UserMapperTest {

    UserMapper mapper;

    @BeforeEach
    public void init(){
        mapper = UserMapper.Companion.getINSTANCE();
    }

    @SneakyThrows
    @Test
    public void UserToEntityTest(){
        var user = new User(new Email("mszewc@edu.pl"), UserRole.ADMIN, "####", true);
        var entity = mapper.mapDomainObjectToEntity(user);
        assertEquals(user.getEmail().getValue(), entity.getEmail());
        assertEquals(user.getPassword(), entity.getPassword());
        assertEquals(user.getRole().toString(), entity.getRole());
        assertEquals(user.getActive(), entity.getActive());
    }

    @Test
    public void EntityToUserTest(){
        var entity = new UserEntity("mszewc@edu.pl", UserRole.ADMIN.toString(), "####", true);
        entity.setGuid(UUID.randomUUID());
        var user = mapper.mapEntityToDomainObject(entity);
        assertEquals(entity.getEmail(), user.getEmail().getValue());
        assertEquals(entity.getPassword(), user.getPassword());
        assertEquals(UserRole.valueOf(entity.getRole()), user.getRole());
        assertEquals(entity.getActive(), user.getActive());
    }

    @SneakyThrows
    @Test
    public void UserToExistingEntityTest(){
        var user = new User(new Email("mszewc@edu.pl"), UserRole.CLIENT, "####", true);
        var entity = new UserEntity("mszewczyk@edu.pl", UserRole.ADMIN.toString(), "wwww", false);
        var guid = UUID.randomUUID();
        entity.setGuid(guid);
        mapper.mapDomainObjectToEntity(user, entity);
        assertEquals(user.getEmail().getValue(), entity.getEmail());
        assertEquals(user.getPassword(), entity.getPassword());
        assertEquals(user.getRole().toString(), entity.getRole());
        assertEquals(user.getActive(), entity.getActive());
        assertEquals(guid, entity.getGuid());
    }

}

File: tks/RepositoriesAdapters/src/test/kotlin/repository/mappers/UserMapperTest.kt
package repository.mappers

import domain.model.User
import domain.model.UserRole
import domain.model.values.Email
import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.assertThrows
import repository.data.UserEntity
import java.util.*

class UserMapperTest {
    private val mapper: UserMapper = UserMapper.INSTANCE

    @Test
    fun userToEntityTest() {
        val user = User(Email("mszewc@edu.pl"), UserRole.ADMIN, "####", true)
        val entity = mapper.mapDomainObjectToEntity(user)
        assertNotNull(entity!!)
        assertEquals(user.email.value, entity.email)
        assertEquals(user.password, entity.password)
        assertEquals(user.role.toString(), entity.role)
        assertEquals(user.active, entity.active)
    }

    @Test
    fun nullableUserToEntity(){
        val entity = mapper.mapDomainObjectToEntity(null)
        assertNull(entity)
    }

    @Test
    fun entityToUserTest() {
        val entity = UserEntity(UUID.randomUUID(), "mszewc@edu.pl", UserRole.ADMIN.toString(), "####", true)
        val user = mapper.mapEntityToDomainObject(entity)
        assertNotNull(user!!)
        assertEquals(entity.email, user.email.value)
        assertEquals(entity.password, user.password)
        assertEquals(UserRole.valueOf(entity.role), user.role)
        assertEquals(entity.active, user.active)
    }

    @Test
    fun nullableEntityToUserTest(){
        val user = mapper.mapEntityToDomainObject(null)
        assertNull(user)
    }

    @Test
    fun userToExistingEntityTest() {
        val user = User(Email("mszewc@edu.pl"), UserRole.CLIENT, "####", true)
        val entity = UserEntity(null, "matzab@edu.pl", UserRole.ADMIN.toString(), "password", false)
        val guid = UUID.randomUUID()
        entity.guid = guid
        mapper.mapDomainObjectToEntity(user, entity)
        assertEquals(user.email.value, entity.email)
        assertEquals(user.password, entity.password)
        assertEquals(user.role.toString(), entity.role)
        assertEquals(user.active, entity.active)
        assertEquals(guid, entity.guid)
    }

    @Test
    fun nullableUserToExistingEntityTest(){
        val entity = UserEntity(null, "matzab@edu.pl", UserRole.ADMIN.toString(), "password", false)
        mapper.mapDomainObjectToEntity(null, entity)
        assertEquals(null, entity.guid)
        assertEquals("matzab@edu.pl", entity.email)
        assertEquals("ADMIN", entity.role)
        assertEquals("password", entity.password)
        assertEquals(false, entity.active)
    }
}